File: qooxdoo-contrib/qwt/trunk/toolkit/compiler/src/test/resources/type/ifc.java
package type;

interface ifc {
    public static final int a = 3;
    long b = 4;
    int foo();
}

class user implements ifc {
    int A = ifc.a;
    long B = ifc.b;
    int AA = a;
    long BB = b;
    
    public int foo() {
        return foo();
    }
    public boolean instanceOf() {
        return this instanceof ifc;
    }
}

File: qooxdoo-contrib/QWT/tags/0.2.0/toolkit/compiler/src/test/resources/type/ifc.js
touchPackage("type");
type.ifc=function() {
}
defineClass(type.ifc, 'type.ifc', null);
type.ifc.a = 0;
type.ifc.b = 0;
type.ifc.prototype.foo = ABSTRACT;
type.ifc.cinit = function() {
type.ifc.a = 3;
type.ifc.b = 4;
}
touchPackage("type");
type.user=function() {
}
defineClass(type.user, 'type.user', java.lang.Object, type.ifc);
type.user.prototype.A = 0;
type.user.prototype.B = 0;
type.user.prototype.AA = 0;
type.user.prototype.BB = 0;
type.user.prototype.foo = function() {
    return this.foo();
}
type.user.prototype.instanceOf = function() {
    return instanceofInterface(this, type.ifc);
}
type.user.cinit = function() {
type.user.prototype.A = type.ifc.a;
type.user.prototype.B = type.ifc.b;
type.user.prototype.AA = type.ifc.a;
type.user.prototype.BB = type.ifc.b;
}
